### PR TITLE
Update mqtt topics

### PR DESCRIPTION
**Pull request by skeeey**
simplify the source/agent client topics, currently, we only need two event topics for source/agent client 

the source client 
- publish its resource spec create/update/delete event or status resync request event with topic  `sources/${SOURCE_ID}/consumers/+/sourceevents`
- subscribe to `$share/statussubscribers/sources/${SOURCE_ID}/consumers/+/agentevents` to receive agent status update event or spec resync request events

the agent client
- publish its resource status update event or spec resync request events with topic  `sources/${SOURCE_ID}/consumers/${CONSUMER_ID}/agentevents`
- subscribe to `sources/${SOURCE_ID}/consumers/${CONSUMER_ID}/sourceevents` to receive source esource spec create/update/delete event or status resync request

The `${SOURCE_ID}` will be `maestro` in production environment

/cc @clyang82 @morvencao @qiujian16

### CODE REVIEW

**Comment by skeeey** [comment]
go clean --testcache
➜  maestro git:(usdkgo) ✗ make test
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -v  \
		./pkg/... \
		./cmd/...
PASS pkg/api.TestJSONMapStausToResourceStatus/empty (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus/content_status (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus/reconcile_status (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus/reconcile_status_with_deleted_Condition (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/empty (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/content_status (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/reconcile_status (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/reconcile_status_with_deleted_Condition (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap (0.00s)
PASS pkg/api
EMPTY pkg/api/openapi
EMPTY pkg/api/presenters
EMPTY pkg/auth
EMPTY pkg/client/cloudevents
EMPTY pkg/client/ocm
PASS pkg/config.TestConfigReadStringFile (0.00s)
PASS pkg/config.TestConfigReadIntFile (0.00s)
PASS pkg/config.TestConfigReadBoolFile (0.00s)
PASS pkg/config.TestConfigReadQuotedFile (0.00s)
PASS pkg/config
PASS pkg/controllers.TestControllerFramework (0.00s)
PASS pkg/controllers
EMPTY pkg/dao
EMPTY pkg/dao/mocks
EMPTY pkg/db
EMPTY pkg/db/db_context
EMPTY pkg/db/db_session
EMPTY pkg/db/migrations
EMPTY pkg/db/mocks
EMPTY pkg/db/transaction
PASS pkg/errors.TestErrorFormatting (0.00s)
PASS pkg/errors.TestErrorFind (0.00s)
PASS pkg/errors
EMPTY pkg/handlers
EMPTY pkg/logger
PASS pkg/services.TestSQLTranslation (0.01s)
PASS pkg/services.TestResourceFindByConsumerID (0.00s)
PASS pkg/services.TestValidateNewManifest/validated (0.00s)
PASS pkg/services.TestValidateNewManifest/no_apiVersion (0.00s)
PASS pkg/services.TestValidateNewManifest/no_kind (0.00s)
PASS pkg/services.TestValidateNewManifest/no_name (0.00s)
PASS pkg/services.TestValidateNewManifest/has_generate_name (0.00s)
PASS pkg/services.TestValidateNewManifest/has_resource_version (0.00s)
PASS pkg/services.TestValidateNewManifest/has_deletion_grace_period_seconds (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_namespace (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_api_version_(no_version) (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_api_version_(no_version)#01 (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_api_version (0.00s)
PASS pkg/services.TestValidateNewManifest (0.00s)
PASS pkg/services.TestValidateUpdateManifest/validated (0.00s)
PASS pkg/services.TestValidateUpdateManifest/apiVersion_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest/kind_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest/name_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest/namespace_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest (0.00s)
PASS pkg/services
EMPTY pkg/util
EMPTY cmd/maestro
EMPTY cmd/maestro/agent
PASS cmd/maestro/environments.TestLoadServices (0.20s)
PASS cmd/maestro/environments
EMPTY cmd/maestro/migrate
EMPTY cmd/maestro/servecmd
EMPTY cmd/maestro/server
EMPTY cmd/maestro/server/logging

DONE 38 tests in 50.810s

go clean --testcache
➜  maestro git:(usdkgo) ✗ make test-integration
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -ldflags -s -v -timeout 1h  \
			./test/integration
I0130 15:18:27.652020    4412 framework.go:76] Initializing testing environment
I0130 15:18:27.941883    4412 framework.go:161] Using Mock OCM Authz Client
I0130 15:18:27.945284    4412 framework.go:199] Disabling Sentry error reporting
{"level":"info","ts":1706599107.945496,"logger":"fallback","caller":"v2@v2.0.0-20231030012137-0836a524e995/protocol.go:124","msg":"subscribing to topics: map[sources/maestro-integration/consumers/+/agentevents:{1 0 false false}]"}
I0130 15:18:27.964426    4412 api_server.go:145] Serving without TLS at localhost:8000
I0130 15:18:27.964622    4412 healthcheck_server.go:56] Serving HealthCheck without TLS at localhost:8083
PASS test/integration.TestConsumerGet (0.17s)
PASS test/integration.TestConsumerPost (0.17s)
PASS test/integration.TestConsumerPaging (0.21s)
PASS test/integration.TestControllerRacing (1.71s)
PASS test/integration.TestControllerReconcile (1.32s)
PASS test/integration.TestControllerSync (1.18s)
PASS test/integration.TestPulseServer (5.33s)
PASS test/integration.TestResourceGet (0.20s)
PASS test/integration.TestResourcePost (2.22s)
PASS test/integration.TestResourcePatch (0.23s)
PASS test/integration.TestResourcePaging (0.30s)
PASS test/integration.TestResourceListSearch (0.29s)
PASS test/integration.TestUpdateResourceWithRacingRequests (0.36s)
I0130 15:18:41.690597    4412 api_server.go:151] Web server terminated
PASS test/integration

DONE 13 tests in 18.076s


make e2e-test
./test/e2e/setup/e2e_teardown.sh
+ command -v kind
+ kind delete cluster --name maestro
enabling experimental podman provider
Deleting cluster "maestro" ...
Deleted nodes: ["maestro-control-plane"]
+ rm -rf ./test/e2e/setup/.kubeconfig
+ rm -rf ./test/e2e/setup/.consumer_id
+ rm -rf ./test/e2e/setup/.external_host_ip
./test/e2e/setup/e2e_setup.sh
+ command -v kind
+ kind create cluster --name maestro --kubeconfig ./test/e2e/.kubeconfig
enabling experimental podman provider
Creating cluster "maestro" ...
 ✓ Ensuring node image (kindest/node:v1.27.3) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-maestro"
You can now use your cluster with:

kubectl cluster-info --context kind-maestro --kubeconfig ./test/e2e/.kubeconfig

Have a nice day! 👋
+ export KUBECONFIG=/home/cloud-user/go/src/github.com/openshift-online/maestro/test/e2e/.kubeconfig
+ KUBECONFIG=/home/cloud-user/go/src/github.com/openshift-online/maestro/test/e2e/.kubeconfig
+ export namespace=maestro
+ namespace=maestro
+ export image_tag=latest
+ image_tag=latest
+ export external_image_registry=image-registry.testing
+ external_image_registry=image-registry.testing
+ export internal_image_registry=image-registry.testing
+ internal_image_registry=image-registry.testing
+ make image
make[1]: Entering directory '/home/cloud-user/go/src/github.com/openshift-online/maestro'
for cmd in $(ls cmd); do \
	go build \
		-ldflags="" \
		-o "${cmd}" \
		"./cmd/${cmd}" \
		|| exit 1; \
done
go: downloading open-cluster-management.io/ocm v0.12.1-0.20240130070023-98ab144e4216
podman build -t "image-registry.testing/maestro/maestro:latest" .
[1/2] STEP 1/7: FROM golang:1.21 AS builder
[1/2] STEP 2/7: ENV SOURCE_DIR=/maestro
--> Using cache 3c70351e42e2d02c10a85c3d14865a4dc49ab2facbe83715bb38273f3a5440fa
--> 3c70351e42e2
[1/2] STEP 3/7: WORKDIR $SOURCE_DIR
--> Using cache d57ebaa7e1c6e9dc2d80a62985b24d416f784575edfd1bf2fff11edefd87bd88
--> d57ebaa7e1c6
[1/2] STEP 4/7: COPY . $SOURCE_DIR
--> 64eba39c4f17
[1/2] STEP 5/7: ENV GOFLAGS=""
--> 88cf560b3cc5
[1/2] STEP 6/7: RUN make binary
...
Successfully tagged image-registry.testing/maestro/maestro:latest
89641823e5389548f45be776517ddcfb9c775e839c9ba03a8a90b1abc79d62f2
make[1]: Leaving directory '/home/cloud-user/go/src/github.com/openshift-online/maestro'
+ kind load docker-image image-registry.testing/maestro/maestro:latest --name maestro
enabling experimental podman provider
Image: "image-registry.testing/maestro/maestro:latest" with ID "89641823e5389548f45be776517ddcfb9c775e839c9ba03a8a90b1abc79d62f2" not yet present on node "maestro-control-plane", loading...
+ kind load docker-image docker.io/library/postgres:14.2 --name maestro
enabling experimental podman provider
Image: "docker.io/library/postgres:14.2" with ID "9dbc24674f25eb449df11179ed3717c47348fb3aa985ae14b3936d54c2c09dde" not yet present on node "maestro-control-plane", loading...
+ kind load docker-image docker.io/library/eclipse-mosquitto:2.0.18 --name maestro
enabling experimental podman provider
Image: "docker.io/library/eclipse-mosquitto:2.0.18" with ID "ff1abc425b332719d56328a2fe99dd2978b08d69d9bd5e53e8bf32e73444b3bb" not yet present on node "maestro-control-plane", loading...
+ kubectl label node maestro-control-plane node-role.kubernetes.io/master=
node/maestro-control-plane labeled
+ kubectl apply -f ./test/e2e/setup/service-ca-crds
customresourcedefinition.apiextensions.k8s.io/clusteroperators.config.openshift.io created
customresourcedefinition.apiextensions.k8s.io/servicecas.operator.openshift.io created
+ kubectl create ns openshift-config-managed
namespace/openshift-config-managed created
+ kubectl apply -f ./test/e2e/setup/service-ca/
clusterrolebinding.rbac.authorization.k8s.io/system:openshift:operator:service-ca-operator created
namespace/openshift-service-ca-operator created
service/metrics created
configmap/service-ca-operator-config created
serviceca.operator.openshift.io/cluster created
serviceaccount/service-ca-operator created
deployment.apps/service-ca-operator created
clusteroperator.config.openshift.io/service-ca created
+ export ENABLE_JWT=false
+ ENABLE_JWT=false
+ export ENABLE_OCM_MOCK=true
+ ENABLE_OCM_MOCK=true
+ export source_id=maestro-development
+ source_id=maestro-development
+ kubectl create namespace maestro
namespace/maestro created
+ make template deploy-secrets deploy-db deploy-mqtt deploy-service
make[1]: Entering directory '/home/cloud-user/go/src/github.com/openshift-online/maestro'
...
+ kubectl apply -n maestro-agent --filename=templates/agent-template.json
+ egrep --color=auto 'configured|$$'
customresourcedefinition.apiextensions.k8s.io/appliedmanifestworks.work.open-cluster-management.io created
clusterrole.rbac.authorization.k8s.io/maestro-agent:agent created
clusterrolebinding.rbac.authorization.k8s.io/maestro-agent:agent created
clusterrolebinding.rbac.authorization.k8s.io/maestro-agent:execution-admin created
clusterrolebinding.rbac.authorization.k8s.io/maestro-agent:execution created
role.rbac.authorization.k8s.io/maestro-agent:agent created
rolebinding.rbac.authorization.k8s.io/maestro-agent:agent created
rolebinding.rbac.authorization.k8s.io/maestro-agent:agent:extension-apiserver created
role.rbac.authorization.k8s.io/maestro-agent:agent:extension-apiserver created
deployment.apps/maestro-agent created
secret/maestro-agent-mqtt created
serviceaccount/maestro-agent-sa created
ginkgo --output-dir="/home/cloud-user/go/src/github.com/openshift-online/maestro/test/e2e/report" --json-report=report.json --junit-report=report.xml \
/home/cloud-user/go/src/github.com/openshift-online/maestro/test/e2e/pkg -- -consumer_id="8aca50a2-fa07-46d5-adf3-b316a9a269d1" \
-api-server=https://10.89.0.3:30080 -kubeconfig=/home/cloud-user/go/src/github.com/openshift-online/maestro/test/e2e/.kubeconfig
Ginkgo detected a version mismatch between the Ginkgo CLI and the version of Ginkgo imported by your packages:
  Ginkgo CLI Version:
    2.5.0
  Mismatched package versions found:
    2.13.2 used by pkg

  Ginkgo will continue to attempt to run but you may see errors (including flag
  parsing errors) and should either update your go.mod or your version of the
  Ginkgo CLI to match.

  To install the matching version of the CLI run
    go install github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file.  Alternatively you can use
    go run github.com/onsi/ginkgo/v2/ginkgo
  from a path that contains a go.mod file to invoke the matching version of the
  Ginkgo CLI.

  If you are attempting to test multiple packages that each have a different
  version of the Ginkgo library with a single Ginkgo CLI that is currently
  unsupported.

Running Suite: End-to-End Test Suite - /home/cloud-user/go/src/github.com/openshift-online/maestro/test/e2e/pkg
===============================================================================================================
Random Seed: 1706598938

Will run 4 of 4 specs
••••

Ran 4 of 4 Specs in 3.065 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 5.745145307s
Test Suite Passed

**Review by qiujian16 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -376,11 +380,11 @@ db/teardown:
 
 .PHONY: mqtt/prepare
 mqtt/prepare:
-	@echo $(shell LC_CTYPE=C && cat /dev/urandom | tr -dc 'a-zA-Z0-9' | head -c 13) > $(mqtt_password_file)
+	@echo $(shell LC_CTYPE=C tr -dc 'a-zA-Z0-9' < /dev/urandom | head -c 13) > $(mqtt_password_file)
 
 .PHONY: mqtt/setup
 mqtt/setup: mqtt/prepare
-	@echo '{"brokerHost": "localhost:1883", "username": "$(mqtt_user)", "password": "$(shell cat $(mqtt_password_file))"}' > $(mqtt_config_file)
+	@echo '{"brokerHost":"localhost:1883","username":"$(mqtt_user)","password":"$(shell cat $(mqtt_password_file))","topics":{"sourceEvents":"sources/maestro-integration/consumers/+/sourceevents","agentEvents":"sources/maestro-integration/consumers/+/agentevents"}}' > $(mqtt_config_file)
```
Comment (morvencao) [inline Makefile:387]: should't we use `$(source_id)`, why hard code `maestro-integration` here?

Comment (skeeey) [reply Makefile:387]: I think this only for integration test, the source id is always `maestro-integration` in  integration test

Comment (skeeey) [reply Makefile:387]: move this config under the `test-integration` task, it may be more better

File: templates/secrets-template.yml [hunk 2]
```diff
@@ -55,6 +55,9 @@ parameters:
 - name: SENTRY_KEY
   description: Private key used in Sentry DSN
 
+- name: SOURCE_ID
+  description: Source id for maestro server maestro server mqtt client.
```
Comment (clyang82) [inline templates/secrets-template.yml:59]: typo error?

File: test/e2e/setup/e2e_setup.sh [hunk 3]
```diff
@@ -42,6 +42,7 @@ kubectl $1 apply -f ./test/e2e/setup/service-ca/
 # 4. deploy maestro into maestro namespace
 export ENABLE_JWT=false
 export ENABLE_OCM_MOCK=true
+export source_id="maestro-development"
```
Comment (clyang82) [inline test/e2e/setup/e2e_setup.sh:45]: need to remove

Comment (skeeey) [reply test/e2e/setup/e2e_setup.sh]: removed

**Pull request merged**